packages/connector-tencent-sms/src/index.ts

```diff
@@ -0,0 +1,120 @@
+import type {
+  CreateConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+  SmsConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { sendSMSRequest } from '@/http';
+import type { TencentSmsConfig } from '@/schema';
+import { SmsConfigGuard, TencentErrorResponse } from '@/schema';
+
+import { defaultMetadata } from './constant';
+
+function sendMessage(getConfig: GetConnectorConfig): SendMessageFunction {
+  // eslint-disable-next-line complexity
+  return async (data, inputConfig) => {
+    const { to, type, payload } = data;
+    const config = inputConfig ?? (await getConfig(defaultMetadata.id));
+    validateConfig<TencentSmsConfig>(config, SmsConfigGuard);
+    const { accessKeyId, accessKeySecret, signName, templates, sdkAppId, region } = config;
+    const template = templates.find(({ usageType }) => usageType === type);
+
+    assert(
+      template,
+      new ConnectorError(
+        ConnectorErrorCodes.TemplateNotFound,
+        `Cannot find template for type: ${type}`
+      )
+    );
+
+    try {
+      const httpResponse = await sendSMSRequest(template.templateCode, [payload.code], to, {
+        secretId: accessKeyId,
+        secretKey: accessKeySecret,
+        sdkAppId,
+        region,
+        signName,
+      });
+
+      const { body: responseData } = httpResponse;
+
+      try {
+        TencentErrorResponse.parse(responseData);
+        throw new HTTPError(httpResponse);
+      } catch {
+        // Not http error
+      }
+
+      const {
+        Response: { SendStatusSet, RequestId },
+      } = responseData;
+
+      // Non-mass sending, match length
+      if (SendStatusSet.length === 1) {
+        const firstElement = SendStatusSet[0];
+
+        if (firstElement) {
+          const { Code, Message } = firstElement;
+
+          if (Code !== 'Ok') {
+            throw new ConnectorError(ConnectorErrorCodes.General, {
+              errorDescription: Message,
+              Code,
+              RequestId,
+            });
+          }
+
+          return httpResponse;
+        }
+
+        throw new ConnectorError(ConnectorErrorCodes.General, {
+          errorDescription: 'SendStatusSet is empty',
+        });
+      }
```

Comment:
same here. Return or throw early instead of nested code blocks. e.g.

```js
const invalidStatusSetError = new ConnectorError(ConnectorErrorCodes.General, {
     errorDescription: 'SendStatusSet is empty',
 });
  
assert(SendStatusSet.length === 1,  invalidStatusSetError);
const firstElement = SendStatusSet[0];
assert(firstElement, invalidStatusSetError);
const { Code, Message } = firstElement;
assert(Code === 'ok', new ConnectorError(ConnectorErrorCodes.General, {
              errorDescription: Message,
              Code,
              RequestId,
            }));
  
 return  httpResponse;
```

Reply:
Here is the rule to avoid `@typescript-eslint/no-non-null-assertion`, which cannot be avoided by using `assert` as well

Reply:
`const firstElement = SendStatusSet[0];` `firstElement` like `SomeType|undefined`